src/client/rewards/Create-Edit/CreateRewardForm.js

```diff
@@ -64,66 +64,52 @@ class CreateRewardForm extends React.Component {
     commissionToWaivio: 5,
     campaignId: null,
     isCampaignActive: false,
-
-    // confirmDirty: false,
-    // autoCompleteResult: [],
-    // isModalEligibleUsersOpen: false,
-    // hasRequireObject: false,
-    // hasReviewObject: false,
   };
 
   componentDidMount = async () => {
     if (this.props.match.params.campaignId) {
+      // eslint-disable-next-line react/no-did-mount-set-state
+      this.setState({ loading: true });
+
       const campaign = await getCampaignById(this.props.match.params.campaignId);
 
-      let primaryObject = await getObjectsByIds({
+      const primaryObject = getObjectsByIds({
         authorPermlinks: [campaign.campaign.requiredObject],
       });
-      primaryObject = getClientWObj(primaryObject.wobjects[0]);
 
-      let secondaryObjects = await getObjectsByIds({
-        authorPermlinks: [...campaign.campaign.objects],
-      });
-      secondaryObjects = secondaryObjects.wobjects.map(obj => getClientWObj(obj));
+      const secondaryObjects = getObjectsByIds({ authorPermlinks: [...campaign.campaign.objects] });
 
-      let sponsors = [];
-      if (!_.isEmpty(campaign.campaign.match_bots)) {
-        sponsors = await getObjectsByIds({ authorPermlinks: [...campaign.campaign.match_bots] });
-        sponsors = sponsors.wobjects.map(obj => getClientWObj(obj));
-      }
       const expiredAt = moment(new Date(campaign.campaign.expired_at));
 
       const isCampaignActive = campaign.campaign.status === 'active';
 
-      console.log(isCampaignActive);
+      let sponsors;
+      if (!isEmpty(campaign.campaign.match_bots)) {
+        sponsors = await getObjectsByIds({ authorPermlinks: [...campaign.campaign.match_bots] });
```

Comment:
There are three the same requests getObjectsByIds() with array as a param.
We can use one instead of three.